backend/test/test_qa.py

```diff
@@ -0,0 +1,167 @@
+from .conftest import API_PREFIX, HOST_PREFIX
+import requests
+import time
+
+qa_run_id = None
+
+
+def test_run_qa(crawler_crawl_id, crawler_auth_headers, default_org_id):
+    r = requests.post(
+        f"{API_PREFIX}/orgs/{default_org_id}/crawls/{crawler_crawl_id}/qa/start",
+        headers=crawler_auth_headers,
+    )
+
+    assert r.status_code == 200
+
+    data = r.json()
+    assert data["started"]
+    global qa_run_id
+    qa_run_id = data["started"]
+
+
+def test_run_qa_already_running(crawler_crawl_id, crawler_auth_headers, default_org_id):
+    r = requests.post(
+        f"{API_PREFIX}/orgs/{default_org_id}/crawls/{crawler_crawl_id}/qa/start",
+        headers=crawler_auth_headers,
+    )
+
+    assert r.status_code == 400
+    assert r.json()["detail"] == "qa_already_running"
+
+
+def test_active_qa(crawler_crawl_id, crawler_auth_headers, default_org_id):
+    r = requests.get(
+        f"{API_PREFIX}/orgs/{default_org_id}/crawls/{crawler_crawl_id}/qa/activeQA",
+        headers=crawler_auth_headers,
+    )
+
+    data = r.json()
+    qa = data["qa"]
+
+    assert qa
+    assert qa["state"]
+    assert qa["started"]
+    assert not qa["finished"]
+
+
+def test_qa_list(crawler_crawl_id, crawler_auth_headers, default_org_id):
+    r = requests.get(
+        f"{API_PREFIX}/orgs/{default_org_id}/crawls/{crawler_crawl_id}/qa",
+        headers=crawler_auth_headers,
+    )
+
+    data = r.json()
+
+    assert len(data) == 1
+
+    qa = data[0]
+    assert qa
+    assert qa["state"]
+    assert qa["started"]
+    assert not qa["finished"]
+
+
+def test_wait_for_complete(crawler_crawl_id, crawler_auth_headers, default_org_id):
+    count = 0
+    completed = False
+    while count < 24:
+        r = requests.get(
+            f"{API_PREFIX}/orgs/{default_org_id}/crawls/{crawler_crawl_id}/qa/activeQA",
+            headers=crawler_auth_headers,
+        )
+
+        data = r.json()
+        if not data["qa"]:
+            completed = True
+            break
+
+        time.sleep(5)
+        count += 1
+
+    assert completed
+
+
+def test_qa_completed(crawler_crawl_id, crawler_auth_headers, default_org_id):
```

Comment:
Could add a test here to check the org stats to make sure the execution time for QA is being updated?

Reply:
Added in https://github.com/webrecorder/browsertrix-cloud/pull/1586/commits/bc3bc4427a802da2d1c40dc4af3b8446cefca3dd as a separate test.